lwt/util.ml

```diff
@@ -0,0 +1,36 @@
+open Xen_api
+open Lwt
+
+let is_ssl ?allow_file:(allow_file=true) uri =
+  match Uri.scheme uri with
+  | Some "https" ->
+    true
+  | Some "http"
+  | Some "file" when allow_file ->
+    false
```

Comment:
This isn't doing what you think it's doing, the when clause it's being applied to the "http" case.
```suggestion
  | Some "http" ->
    false
  | Some "file" when allow_file ->
    false
```